nx/guides/getting_started/introduction.md

```diff
@@ -0,0 +1,82 @@
+# What is Nx?
```

Comment:
@josevalim This is the start of our revamped docs.

We're taking the current getting started guide and both splitting it and getting into more detail.
I'm thinking we should merge these onto a `new-docs` branch and only merge that onto main after the getting started is fully done.

WDYT?

Reply:
I think it is fine to push to main directly, given the plan is for continuous work on it, right?

Reply:
We'll make it so that this PR is merged when it fully replaces the previous introduction to nx guide.